<?xml version="1.0"?>

<launch>
	<param name="vehicle/scripts_dir" type="str" value="$(find mkz_mpc_path_follower)/scripts/" />
  <!-- The trajectory can be generated using recorded time or a target velocity. -->
  <!-- Set track_using_time to True to allow for time-varying velocity. -->
  <!-- Set track_using_time to False to use target_vel (m/s) instead. -->
  <param name="vehicle/track_using_time" type="bool" value="True" />
  <param name="vehicle/target_vel" type="double" value="1.0" />

  <!-- Waypoints CSV to specify path. -->
  <!-- Set is_heading_info to true if Heading (N = 0) is used instead of Yaw (E = 0). -->
  <param name="vehicle/mat_waypoints" type="str" value="$(find mkz_mpc_path_follower)/paths/path3_6_20.mat" />
  <param name="vehicle/is_heading_info" type="bool" value="True"/>
  <param name="vehicle/time_check_on" type="bool" value="False"/>

  <!-- RFS Coordinate System Definition -->
  <param name="vehicle/lat0" type="double" value="37.917929"/>
  <param name="vehicle/lon0" type="double" value="-122.331798"/>
  <param name="vehicle/yaw0" type="double" value="0.0"/>

  <!-- Vehicle Simulation Initial Conditions: Good for Path 1/3, Path Lot Stop Sign -->
  <param name="vehicle/X0" type="double" value="0.0"/>
  <param name="vehicle/Y0" type="double" value="3.0"/>
  <param name="vehicle/Psi0" type="double" value="-1.5"/>

  <!-- Vehicle Simulation Initial Conditions: Good for Path 2, RFS Gate -->
  <!--param name="vehicle/X0" type="double" value="142.0"/>
  <param name="vehicle/Y0" type="double" value="-82.0"/>
  <param name="vehicle/Psi0" type="double" value="2.0"/-->

  <!--...................................................................-->

  <!-- MPC, Simulation, and Plotting Nodes -->
  <node ns="vehicle" pkg="mkz_mpc_path_follower" type="mpc_cmd_pub.jl" name="mpc_node" output="log"/>
  <node ns="vehicle" pkg="mkz_mpc_path_follower" type="vehicle_simulator.py" name="sim_node" output="log"/>
  <node ns="vehicle" pkg="mkz_mpc_path_follower" type="gps_plotter.py" name="plot_node" output="log"/>

  <!-- Recording Node:: TODO FIX TOPICS -->
  <arg name="can_topics" default=""/>
  <arg name="gps_topics" default=""/>
  <arg name="vehicle_topics" default="/vehicle/state_est /vehicle/target_path /vehicle/mpc_path"/>
  <arg name="control_topics" default=""/>
  <arg name="perception_topics" default=""/>

  <node pkg="rosbag" type="record" name="rosbag_path_recorder" respawn="true" output="screen" 
        args="-o $(find mkz_mpc_path_follower)/bags/path_sim $(arg can_topics) $(arg gps_topics) $(arg vehicle_topics) $(arg control_topics) $(arg perception_topics)" />

</launch>
